app/scripts/metamask-controller.js

```diff
@@ -2673,6 +2680,18 @@ export default class MetamaskController extends EventEmitter {
     }
   }
 
+  /**
+   * Sets the Ledger Live preference to use for hardware wallet support
+   * @param {bool} bool - the value representing if the users wants to use Ledger Live
+   */
+  async setLedgerLivePreference(bool) {
+    this.preferencesController.setLedgerLivePreference(bool);
+    const keyring = await this.getKeyringForDevice('ledger');
+    if (keyring?.updateTransportMethod) {
+      keyring.updateTransportMethod(bool);
```

Comment:
If this call fails, the UI would show no indication of failure. What's worse, the toggle state would not reflect the keyring state.

Ideally we'd `await` this call, so that we could forward the error to the UI. Ideally we'd try to keep it in-sync with the preference as well (e.g. maybe add a `catch` block where the previous preference value is restored? Or maybe set the preference only of this call succeeds? Something like that).

Reply:
I've added the fallback.  Unfortunately, however, we should not `await` this because it's also called with `submitPassword`, and would delay the entire first metamask login/load until GitHub was happy.